vcpkg can't be used on a system with only VS2015 installed

I cannot install any library, not even the one at the readme page

>PS C:\Development Softwares\vcpkg> .\vcpkg install sdl2 curl
Computing installation plan...
The following packages will be built and installed:
    curl[core,non-http,ssl,sspi,winssl]:x86-windows
    sdl2[core]:x86-windows
>  * zlib[core]:x86-windows
Additional packages (*) will be modified to complete this operation.
Starting package 1/3: zlib:x86-windows
Building package zlib[core]:x86-windows...
-- Using cached C:/Development Softwares/vcpkg/downloads/zlib1211.tar.gz
-- Extracting source C:/Development Softwares/vcpkg/downloads/zlib1211.tar.gz
-- Applying patch cmake_dont_build_more_than_needed.patch
-- Using source at C:/Development Softwares/vcpkg/buildtrees/zlib/src/1.2.11-deec42f53b
-- Configuring x86-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: ninja -v
    Working Directory: C:/Development Softwares/vcpkg/buildtrees/zlib/x86-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\Development Softwares\vcpkg\buildtrees\zlib\config-x86-windows-out.log

>Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:297 (vcpkg_execute_required_process)
  ports/zlib/portfile.cmake:22 (vcpkg_configure_cmake)
  scripts/ports.cmake:90 (include)


>Error: Building package zlib:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: zlib:x86-windows
  Vcpkg version: 2020.02.04-nohash

>[1/2] cmd /c "cd .. && "C:/Development Softwares/vcpkg/downloads/tools/cmake-3.17.2-windows/cmake-3.17.2-win32-x86/bin/cmake.exe" "C:/Development Softwares/vcpkg/buildtrees/zlib/src/1.2.11-deec42f53b" "-DSKIP_INSTALL_FILES=ON" "-DSKIP_BUILD_EXAMPLES=ON" "-DCMAKE_MAKE_PROGRAM=C:/Development Softwares/vcpkg/downloads/tools/ninja/ninja-1.10.0/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/Development Softwares/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v140" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/Development Softwares/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=C:/Development Softwares/vcpkg/packages/zlib_x86-windows""
FAILED: ../CMakeCache.txt 
cmd /c "cd .. && "C:/Development Softwares/vcpkg/downloads/tools/cmake-3.17.2-windows/cmake-3.17.2-win32-x86/bin/cmake.exe" "C:/Development Softwares/vcpkg/buildtrees/zlib/src/1.2.11-deec42f53b" "-DSKIP_INSTALL_FILES=ON" "-DSKIP_BUILD_EXAMPLES=ON" "-DCMAKE_MAKE_PROGRAM=C:/Development Softwares/vcpkg/downloads/tools/ninja/ninja-1.10.0/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/Development Softwares/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v140" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/Development Softwares/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=C:/Development Softwares/vcpkg/packages/zlib_x86-windows""
CMake Warning at C:/Development Softwares/vcpkg/scripts/buildsystems/vcpkg.cmake:157 (message):
  There are no libraries installed for the Vcpkg triplet x86-windows.
Call Stack (most recent call first):
  C:/Development Softwares/vcpkg/downloads/tools/cmake-3.17.2-windows/cmake-3.17.2-win32-x86/share/cmake-3.17/Modules/CMakeDetermineSystem.cmake:93 (include)
  CMakeLists.txt:4 (project)


>-- The C compiler identification is unknown
CMake Error at CMakeLists.txt:4 (project):
  No CMAKE_C_COMPILER could be found.

>  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.


>-- Configuring incomplete, errors occurred!
See also "C:/Development Softwares/vcpkg/buildtrees/zlib/x86-windows-rel/CMakeFiles/CMakeOutput.log".
See also "C:/Development Softwares/vcpkg/buildtrees/zlib/x86-windows-rel/CMakeFiles/CMakeError.log".
[2/2] cmd /c "cd ../../x86-windows-dbg && "C:/Development Softwares/vcpkg/downloads/tools/cmake-3.17.2-windows/cmake-3.17.2-win32-x86/bin/cmake.exe" "C:/Development Softwares/vcpkg/buildtrees/zlib/src/1.2.11-deec42f53b" "-DSKIP_INSTALL_FILES=ON" "-DSKIP_BUILD_EXAMPLES=ON" "-DCMAKE_MAKE_PROGRAM=C:/Development Softwares/vcpkg/downloads/tools/ninja/ninja-1.10.0/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/Development Softwares/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v140" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/Development Softwares/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-DSKIP_INSTALL_HEADERS=ON" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=C:/Development Softwares/vcpkg/packages/zlib_x86-windows/debug""
FAILED: ../../x86-windows-dbg/CMakeCache.txt 
cmd /c "cd ../../x86-windows-dbg && "C:/Development Softwares/vcpkg/downloads/tools/cmake-3.17.2-windows/cmake-3.17.2-win32-x86/bin/cmake.exe" "C:/Development Softwares/vcpkg/buildtrees/zlib/src/1.2.11-deec42f53b" "-DSKIP_INSTALL_FILES=ON" "-DSKIP_BUILD_EXAMPLES=ON" "-DCMAKE_MAKE_PROGRAM=C:/Development Softwares/vcpkg/downloads/tools/ninja/ninja-1.10.0/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/Development Softwares/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v140" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/Development Softwares/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-DSKIP_INSTALL_HEADERS=ON" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=C:/Development Softwares/vcpkg/packages/zlib_x86-windows/debug""
CMake Warning at C:/Development Softwares/vcpkg/scripts/buildsystems/vcpkg.cmake:157 (message):
  There are no libraries installed for the Vcpkg triplet x86-windows.
Call Stack (most recent call first):
  C:/Development Softwares/vcpkg/downloads/tools/cmake-3.17.2-windows/cmake-3.17.2-win32-x86/share/cmake-3.17/Modules/CMakeDetermineSystem.cmake:93 (include)
  CMakeLists.txt:4 (project)


>-- The C compiler identification is unknown
CMake Error at CMakeLists.txt:4 (project):
  No CMAKE_C_COMPILER could be found.

>  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.


>-- Configuring incomplete, errors occurred!
See also "C:/Development Softwares/vcpkg/buildtrees/zlib/x86-windows-dbg/CMakeFiles/CMakeOutput.log".
See also "C:/Development Softwares/vcpkg/buildtrees/zlib/x86-windows-dbg/CMakeFiles/CMakeError.log".
ninja: build stopped: subcommand failed.

I have visual studio 2015 update 3 installed, cmake runs perfectly outside of vcpkg.

>PS C:\Development Softwares\vcpkg\downloads\tools\cmake-3.17.2-windows\cmake-3.17.2-win32-x86\bin> .\cmake.exe "C:\Development Softwares\vcpkg\buildtrees\bzip2\src\1.0.6-8838f21f8e"
-- Building for: Visual Studio 14 2015
-- Selecting Windows SDK version  to target Windows 10.0.17763.
-- The C compiler identification is MSVC 19.0.24215.1
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/cl.exe - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Development Softwares/vcpkg/downloads/tools/cmake-3.17.2-windows/cmake-3.17.2-win32-x86/bin

It just cannot find the compiler at all when run from inside vcpkg

--- a/toolsrc/src/vcpkg/metrics.cpp
+++ b/toolsrc/src/vcpkg/metrics.cpp
@@ -32,7 +32,7 @@ namespace vcpkg::Metrics
 
     struct append_hexits {
         constexpr static char hex[17] = "0123456789abcdef";
-        void operator()(std::string& res, std::uint8_t bits) {
+        void operator()(std::string& res, std::uint8_t bits) const {
             res.push_back(hex[(bits >> 4) & 0x0F]);
             res.push_back(hex[(bits >> 0) & 0x0F]);
         }

--- a/toolsrc/src/vcpkg/metrics.cpp
+++ b/toolsrc/src/vcpkg/metrics.cpp
@@ -30,6 +30,14 @@ namespace vcpkg::Metrics
         return "";
     }
 
+    struct append_hexits {
+        constexpr static char hex[17] = "0123456789abcdef";
+        void operator()(std::string& res, std::uint8_t bits) {
+            res.push_back(hex[(bits >> 4) & 0x0F]);
+            res.push_back(hex[(bits >> 0) & 0x0F]);
+        }
+    };
+
     // note: this ignores the bits of these numbers that would be where format and variant go
     static std::string uuid_of_integers(uint64_t top, uint64_t bottom)
     {
@@ -40,11 +48,7 @@ namespace vcpkg::Metrics
         // uuid_field_size in hex characters, not bytes
         constexpr size_t uuid_size = 8 + 1 + 4 + 1 + 4 + 1 + 4 + 1 + 12;
 
-        constexpr static char hex[17] = "0123456789abcdef";
-        constexpr static auto write_byte = [](std::string& res, std::uint8_t bits) {
-            res.push_back(hex[(bits >> 4) & 0x0F]);
-            res.push_back(hex[(bits >> 0) & 0x0F]);
-        };
+        constexpr static append_hexits write_byte;
 
         // set the version bits to 4
         top &= 0xFFFF'FFFF'FFFF'0FFFULL;

